### ISSUE TITLE
ecmaVersion does not matter in parserOptions ?

### ISSUE THREAD

**Issue by Mavrick91** [eslint/espree#427]
Hi, i am trying to understand how espree works and do some 'test'

but it appear when i use
`var espree = require('espree');`
`var code = 'async function foo() {}';`
`console.log(espree.parse(code));`

with 0 configuration (everything to default), it doesn't trigger an error ?

if i understand it correctly this `async/await` is from ES8 so why is this working ?

Thank you !

**Comment by mysticatea** [comment]
Thank you for your report.

However, I could not reproduce it. Would you provide full repro steps?

```js
// test.js
const espree = require("espree")
const code = "async function foo() {}"
console.log(espree.parse(code))
```

```
> node test
C:\Users\t-nagashima.AD\dev\sandbox\node_modules\espree\lib\espree.js:258
        throw err;
        ^

SyntaxError: Unexpected token function
    at Espree.raise (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\lib\espree.js:253:21)
    at Espree.unexpected (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\lib\espree.js:298:14)
    at Espree.pp.semicolon (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\node_modules\?[4macorn?[24m\dist\acorn.js:666:66)
    at Espree.pp$1.parseExpressionStatement (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\node_modules\?[4macorn?[24m\dist\acorn.js:1145:10)
    at Espree.pp$1.parseStatement (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\node_modules\?[4macorn?[24m\dist\acorn.js:880:26)
    at Espree.pp$1.parseTopLevel (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\node_modules\?[4macorn?[24m\dist\acorn.js:746:23)
    at Espree.parseTopLevel (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\lib\espree.js:241:22)
    at Espree.parse (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\node_modules\?[4macorn?[24m\dist\acorn.js:553:17)
    at Espree.parse (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\lib\espree.js:207:31)
    at Object.parse (C:\Users\t-nagashima.AD\dev\sandbox\node_modules\?[4mespree?[24m\espree.js:133:38) {
  index: ?[33m6?[39m,
  lineNumber: ?[33m1?[39m,
  column: ?[33m7?[39m
}
```

**Comment by Mavrick91** [comment]
Okay when I tried to reproduce the error, I finally understood how eslint worked !

I didn't know when you extend a config (ex: airbnb) it could setup everything, I just thought it was only rules .. !

Thank you ! ;)
